src/pages/concept-list-page/concept-list-page.ts

```diff
@@ -3,18 +3,22 @@ import { compose, withProps } from 'recompose';
 
 import { ConceptListPagePure } from './concept-list-page-pure';
 import { conceptListResolver } from './concept-list-resolver';
-import { requirePermissionDecorator } from '../../auth/require-permission-decorator';
-import { PERMISSION_READ } from '../../auth/auth-service';
+import { hasPermissionForResource, logout, PERMISSION_READ } from '../../auth/auth-service';
+import { requirePredicate } from '../../lib/require-predicate';
 
 const mapRouteParams = withProps(({ match: { params } }) => _.pick(params, 'catalogId'));
 
 const enhance = compose(
   mapRouteParams,
-  requirePermissionDecorator({
-    resource: 'organization',
-    resourceIdPropName: 'catalogId',
-    permission: PERMISSION_READ
-  }),
+  requirePredicate(
+    ({ catalogId }) =>
+      hasPermissionForResource({
+        resource: 'organization',
+        resourceId: catalogId,
+        permission: PERMISSION_READ
+      }),
+    () => logout()
```

Comment:
How about just sending logout as a reference?

Reply:
this is temporary refactoring, better implementation in next commit